src/domain/flow-system/services/src/flow/flow_service_impl.rs

```diff
@@ -625,31 +676,39 @@ impl FlowServiceImpl {
         trigger: FlowTrigger,
         config_snapshot: Option<FlowConfigurationSnapshot>,
     ) -> Result<Flow, InternalError> {
+        let flow_event_store = self.catalog.get_one::<dyn FlowEventStore>().unwrap();
```

Comment:
And in this place it's not very clear from which catalog we are taking the store from. If it's not from transactional catalog, will this even work?